docs/reference/content/upgrading.md

```diff
@@ -7,11 +7,39 @@ title = "Upgrade Considerations"
   pre = "<i class='fa fa-level-up'></i>"
 +++
 
-## Upgrading from 3.x
-
-... TODO ...
-
-## Upgrading from mongo-java-driver-reactivestreams 1.12
+# Upgrading to the 4.0 Driver
+
+## Upgrading from the 3.12 Java driver
+
+The 4.0 release is a major release as per the definition of semantic versioning included in [semver.org]. As such, users
+that upgrade to this release should expect breaking changes. That said, we have attempted to ensure that the upgrade 
+process is as seamless as possible.  Breaking changes are as follows:
+
+  * Numerous classes and methods have been removed from the driver. All of these API elements are annotated as deprecated in the 
```

Comment:
The main thing to look at is whether we need more detail here.  Is a bulleted list sufficient, or do we need space for examples, etc?